snorkel/labeling/model/wrapper.py

```diff
@@ -0,0 +1,226 @@
+from typing import Any, List, Optional
+
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.model_selection import PredefinedSplit
+
+from snorkel.labeling import LabelModel
+from snorkel.utils.lr_schedulers import LRSchedulerConfig
+from snorkel.utils.optimizers import OptimizerConfig
+
+
+class SklearnLabelModel(BaseEstimator, ClassifierMixin):
+    """A sklearn wrapper for LabelModel for using sklearn GridSearch, RandomSearch.
+
+    Uses output of create_param_search_data.
+
+    Note that all hyperparameters for the fit and score functions are accepted at the time the class is defined.
+
+    Examples
+    --------
+    >>> L_train = np.array([[1, 1, 1, 0], [0, 1, 1, 0], [1, 0, 0, 1], [0, 1, 0, 1], [0, 0, 0, 1]])
+    >>> L_dev = np.array([[1, 1, 1, 0], [0, 1, 1, 0], [1, 0, 0, 1], [0, 1, 0, 1]])
+    >>> Y_dev = np.array([1, 1, 0, 1])
+
+    >>> from sklearn.model_selection import GridSearchCV
+    >>> param_grid = [{"lr": [0.01, 0.0000001],
+    ...    "l2": [0.001, 0.5], "metric": ["accuracy"]}]
+    >>> wrapper = SklearnLabelModel()
+    >>> L, Y, cv_split = wrapper.create_param_search_data(L_train, L_dev, Y_dev)
+    >>> clf = GridSearchCV(wrapper, param_grid, cv=cv_split)
+    >>> clf.fit(L, Y)  # doctest: +SKIP
+    >>> clf.best_score_  # doctest: +SKIP
+
+    >>> from sklearn.model_selection import RandomizedSearchCV
+    >>> param_dist = {"lr": np.random.uniform(low=1e-7, high=1e-2, size=(50,)),
+    ...     "l2": np.random.uniform(low=1e-2, high=0.5, size=(50,))}
+    >>> random_search = RandomizedSearchCV(wrapper, param_distributions=param_dist, n_iter=4, cv=cv_split, iid=False)
+    >>> random_search.fit(L, Y)  # doctest: +SKIP
+    >>> random_search.best_score_  # doctest: +SKIP
```

Comment:
let's print the output of this right?

Reply:
using `np.around` since this is random search and best score can be 0.75 or 0.1